src/threat_model/rust_client_threat_model.md

```diff
@@ -0,0 +1,144 @@
+the configuration data and the application identity, as the identity is stored within the client
+library and could thus be considered part of its configuration data.
+
+**Confidentiality** : if known (the authentication token), an attacker could impersonate a specific
+application and execute operations in their name.
+
+**Integrity** : the data should only be modified by the client application as doing otherwise might
+prevent the client application from accessing Parsec functionality.
+
+### Client’s cryptographic keys - AS4
+
+Keys provisioned by the client within the service.
+
+**Confidentiality** : (private) key material is sensitive data and should not be exposed.
+
+**Integrity** : neither public nor private key data should be modifiable by attackers.
+
+**Availability** : the client must always be able to use their cryptographic material.
+
+### System and client application stability - AS5
+
+The application which is using the Parsec client library and the sytem as a whole must be kept
+stable and responsive.
+
+**Availability** : the client application must not be crashed by the library in any way.
+
+## Attackers
+
+Each dataflow is analysed from an attacker’s perspective using STRIDE method. Nothing is supposed
+on the different components.
+
+In the following tables are present the type of each possible threat, its description, its
+mitigation status and the assets impacted. A threat can be unmitigated (U), mitigated (M) or
+operational (O). The assumptions context applies for all threats but when one of them is
+particularly relevant, it will be noted with ASUM.
+
+### Attacker “Request to Service” - A1
+
+This attacker uses the inter-process endpoint, presumably created by the service, to interfere with
+the communication between client and service.
+
+|   | Description                                                                                                                                       | Mitigation    | Assets        |
+|---|---------------------------------------------------------------------------------------------------------------------------------------------------|---------------|---------------|
+| S | An attacker impersonates the service by establishing an IPC endpoint that spoofs the one used by Parsec to exfiltrate any data found in requests. | O-0, O-1      | AS1, AS2, AS4 |
+| S | An attacker impersonates the service and responds to requests using bogus or malicious data.                                                      | O-0, O-1      | AS2           |
+| T | The attacker modifies the data sent or received by the client while it is in transit through the IPC.                                             | M-0           | AS2, AS4      |
+| R | The service denies having sent responses to or received some requests from the client.                                                            | U-1, O-3      |               |
```

Comment:
For `O-3`, maybe I have said something wrong 😅 Since in this TM the service in untrusted, maybe the mitigation for this should be that the **client** logs all requests sent and response received as opposed as the service?

Reply:
But the client logs don't really help with that - you know the client sent that since you're using the library, but you need "proof" that it was indeed the service that handled the request.

Reply:
I see. I guess that implies that the service currently used is trusted.

Reply:
Yeah, after the log happens (and if configured to log everything), the service can't deny it handled said request.